src/highdicom/legacy/sop.py

```diff
@@ -25,384 +36,2630 @@
+            v22 = [v2]
+        else:
+            v11 = v1
+            v22 = v2
+        if len(v11) != len(v22):
+            return False
+        for xx, yy in zip(v11, v22):
+            if isinstance(xx, DSfloat) or isinstance(xx, float):
+                if not is_equal_float(xx, yy):
+                    return False
+            else:
+                if xx != yy:
+                    return False
+        return True
+
+    @classmethod
+    def isequal_dicom_dataset(cls, ds1: Dataset, ds2: Dataset) -> bool:
+        """Checks if two dicom dataset have the same value in all attributes
+        Parameters
+        ----------
+        ds1: pydicom.dataset.Dataset
+            1st dicom dataset
+        ds2: pydicom.dataset.Dataset
+            2nd dicom dataset
+        Returns
+        -------
+        True of dicom datasets are equal otherwise False
+        """
+
+        if type(ds1) != type(ds2):
+            return False
+        if not isinstance(ds1, Dataset):
+            return False
+        for k1, elem1 in ds1.items():
+            if k1 not in ds2:
+                return False
+            elem2 = ds2[k1]
+            if not cls.isequal(elem2.value, elem1.value):
+                return False
+        return True
+
+    @classmethod
+    def tag2str(cls, tg: BaseTag) -> str:
+        """Converts tag to hex format like (group, element)
+
+        """
+
+        if not isinstance(tg, BaseTag):
+            tg = Tag(tg)
+        return '(0x{:0>4x}, 0x{:0>4x})'.format(tg.group, tg.element)
+
+    @classmethod
+    def tag2kwstr(cls, tg: BaseTag) -> str:
+        """Converts tag to keyword and (group, element) form
+
+        """
+
+        return '{}-{:32.32s}'.format(
+            cls.tag2str(tg), keyword_for_tag(tg))
+
+
+class GeometryOfSlice:
+
+    """A class for checking dicom slices geometry/parallelity"""
+
+    def __init__(self,
+                 row_vector: ndarray,
+                 col_vector: ndarray,
+                 top_left_corner_pos: ndarray,
+                 voxel_spacing: ndarray,
+                 dimensions: tuple) -> None:
+        """
+        Parameters
+        ----------
+        row_vector: numpy.ndarray
+            3D vector representing row of the input slice
+        col_vector: numpy.ndarray
+            3D vector representing column the input slice
+        top_left_corner_pos: numpy.ndarray
+            3D point representing top left corner position of the input slice
+        voxel_spacing: numpy.ndarray
+            Three element array. 1st and 2nd copied from PixelSpacing and the
+            3rd copied from SliceThickness
+        dimensions: tuple
+            3 element tuple holding x as number of cols, y as number of rows
+            and z as 1
+        """
+
+        self.row_vector = row_vector
+        self.col_vector = col_vector
+        self.top_left_corner_position = top_left_corner_pos
+        self.voxel_spacing = voxel_spacing
+        self.Dim = dimensions
```

Comment:
```suggestion
        self.dim = dimensions
```
Capitalisation. But also this attribute seems to be unused so maybe we can just delete it? In that case the `dimensions` parameter to the `__init__` would also be unused.

Reply:
Agreed.